Microsoft.HealthVault.Fhir/Transformers/ImmunizationToHealthVault.cs

```diff
@@ -0,0 +1,118 @@
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using Hl7.Fhir.Support;
+using Microsoft.HealthVault.ItemTypes;
+using System.Linq;
+using Hl7.Fhir.Model;
+using System;
+using Microsoft.HealthVault.Fhir.Constants;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static class ImmunizationToHealthVault
+    {
+        public static ItemTypes.Immunization ToHealthVault(this Hl7.Fhir.Model.Immunization fhirImmunization)
+        {
+            ItemTypes.Immunization hvImmunization = fhirImmunization.ToThingBase<ItemTypes.Immunization>();
+
+            hvImmunization.Name = fhirImmunization.VaccineCode.ToCodableValue();
+            hvImmunization.DateAdministrated = fhirImmunization.DateElement.ToAproximateDateTime();
+
+            if (!fhirImmunization.Practitioner.IsNullOrEmpty())
+            {
+                hvImmunization.Administrator = GetProvider(fhirImmunization);
+            }
+
+            if (!fhirImmunization.Manufacturer.IsNullOrEmpty())
+            {
+                hvImmunization.Manufacturer = GetManufacturer(fhirImmunization);
+            }
+
+            hvImmunization.Lot = fhirImmunization.LotNumber;
+            hvImmunization.Route = fhirImmunization.Route.ToCodableValue();
+            if (fhirImmunization.ExpirationDateElement != null)
+            {
+                hvImmunization.ExpirationDate = fhirImmunization.ExpirationDateElement.ToAproximateDateTime().ApproximateDate;
+            }
+            hvImmunization.AnatomicSurface = fhirImmunization.Site.ToCodableValue();
+
+            Extension immunizationExtension = fhirImmunization.GetExtension(HealthVaultExtensions.ImmunizationDetail);
+            if (immunizationExtension != null)
+            {
+                hvImmunization.Sequence = immunizationExtension.GetStringExtension(HealthVaultExtensions.ImmunizationDetailSequence);
+                hvImmunization.Consent = immunizationExtension.GetStringExtension(HealthVaultExtensions.ImmunizationDetailConcent);
+                hvImmunization.AdverseEvent = immunizationExtension.GetStringExtension(HealthVaultExtensions.ImmunizationDetailAdverseEvent);
+            }
+            if (!fhirImmunization.Note.IsNullOrEmpty())
+            {
+                fhirImmunization.Note.ForEach(note =>
+                {
+                    string sepearator = string.Empty;
+                    if (!string.IsNullOrEmpty(hvImmunization.CommonData.Note))
+                    {
+                        sepearator = "\n"; //Let's seperate each note with new line
```

Comment:
Is this being used anywhere? Also, shouldn't we just use `Environment.NewLine` instead of a custom variable?